src/Core/Services/Cache/DabCacheService.cs

```diff
@@ -78,10 +78,43 @@ public DabCacheService(IFusionCache cache, ILogger<DabCacheService>? logger, IHt
         return result;
     }
 
+    /// <summary>
+    /// Attempts to fetch response from cache. If there is a cache miss, invoke executeQueryAsync Func to get a response
+    /// </summary>
+    /// <typeparam name="TResult">Response payload Type</typeparam>
+    /// <param name="executeQueryAsync">Func with a result of type TResult. Only executed after a cache miss.</param>
+    /// <param name="queryMetadata">Metadata used to create a cache key or fetch a response from the database.</param>
+    /// <param name="cacheEntryTtl">Number of seconds the cache entry should be valid before eviction.</param>
+    /// <returns>JSON Response</returns>
+    /// <exception cref="Exception">Throws when the cache-miss factory method execution fails.</exception>
+    public async ValueTask<TResult?> GetOrSetAsync<TResult>(Func<Task<TResult>> executeQueryAsync, DatabaseQueryMetadata queryMetadata, int cacheEntryTtl)
+    {
+
+        string cacheKey = CreateCacheKey(queryMetadata);
+        TResult? result = await _cache.GetOrSetAsync(
+               key: cacheKey,
+               async (FusionCacheFactoryExecutionContext<TResult> ctx, CancellationToken ct) =>
+               {
+                   TResult result = await executeQueryAsync();
+
+                   ctx.Options.SetSize(EstimateCacheEntrySize(cacheKey: cacheKey, cacheValue: JsonSerializer.Serialize(result?.ToString())));
+                   ctx.Options.SetDuration(duration: TimeSpan.FromSeconds(cacheEntryTtl));
+
+                   return result;
+
+               });
+
+        return result;
+    }
+
     /// <summary>
     /// Creates a cache key using the request metadata resolved from a built SqlQueryStructure
     /// Format: DataSourceName:QueryText:JSON_QueryParameters
     /// Example: 7a07f92a-1aa2-4e2a-81d6-b9af0a25bbb6:select * from MyTable where id = @param1 = :{"@param1":{"Value":"42","DbType":null}}
+    /// Format CosmosDB: DataSourceName:IsPaginated:ContinuationToken:QueryText:JSON_QueryParameters
```

Comment:
why do we need bool `IsPaginated` in the cache key?

Reply:
removed the check.